packages/access-client/src/agent.js

```diff
@@ -409,6 +487,185 @@ export class Agent {
     }
   }
 
+  /**
+   * Request authorization of a session allowing this agent to issue UCANs
+   * signed by the passed email address.
+   *
+   * @param {`${string}@${string}`} email
+   * @param {object} [opts]
+   * @param {AbortSignal} [opts.signal]
+   * @param {Iterable<{ can: Ucanto.Ability }>} [opts.capabilities]
+   */
+  async authorize(email, opts) {
```

Comment:
We should make this a use-case fn that operates on agent.

We can do it if we do the same for `this.#waitForDelegation`

Reply:
https://github.com/web3-storage/w3protocol/pull/563